internal/model/modelindexer/indexer.go

```diff
@@ -101,25 +112,52 @@ type Config struct {
 	// MaxRequests holds the maximum number of bulk index requests to execute concurrently.
 	// The maximum memory usage of Indexer is thus approximately MaxRequests*FlushBytes.
 	//
-	// If MaxRequests is less than or equal to zero, the default of 25 will be used.
+	// If MaxRequests is less than or equal to zero, the default of 50 will be used.
 	MaxRequests int
 
 	// FlushBytes holds the flush threshold in bytes. If Compression is enabled,
 	// The number of events that can be buffered will be greater.
 	//
-	// If FlushBytes is zero, the default of 2MB will be used.
+	// If FlushBytes is zero, the default of 1MB will be used.
 	FlushBytes int
 
 	// FlushInterval holds the flush threshold as a duration.
 	//
 	// If FlushInterval is zero, the default of 30 seconds will be used.
 	FlushInterval time.Duration
 
-	// Tracer holds an optional apm.Tracer to use for tracing bulk requests
-	// to Elasticsearch. Each bulk request is traced as a transaction.
+	// Scaling configuration for the modelindexer.
 	//
-	// If Tracer is nil, requests will not be traced.
-	Tracer *apm.Tracer
+	// If unset, scaling is enabled by default.
+	Scaling ScalingConfig
+}
+
+// ScalingConfig holds the modelindexer scaling configuration.
+type ScalingConfig struct {
+	// Disabled toggles active indexer scaling on.
+	//
+	// It is enabled by default.
+	Disabled bool
+
+	// DownScale holds the scale down config.
+	DownScale ScaleActionConfig
+
+	// ScaleUpThreshold holds the scale up config.
+	UpScale ScaleActionConfig
+
+	// IdleInterval defines how long an active indexer performs an inactivity
+	// check based on the DownScale settings.
+	IdleInterval time.Duration
+}
+
+// ScaleActionConfig holds the configuration for a scaling action
+type ScaleActionConfig struct {
+	// Threshold is the number on which the scaling action will be triggered.
```

Comment:
The number of what? Could you add some comments to the ScalingConfig.DownScale and ScalingConfig.UpScale fields to describe what the thresholds are in terms of? Alternatively, maybe we should fold these fields directly into ScalingConfig and come up with less abstract names. I couldn't think of any off the top of my head though.